service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -863,15 +872,41 @@ public UserProfileDto getProfileData(String uuid) {
         User user = userRepository.findByUuid(uuid);
         List<Address> allAddress = addressRepo.findAllByUserId(user.getId());
         UserProfileDto userProfileDto = modelMapper.map(user, UserProfileDto.class);
+        List<Bot> botList = getListOfBots(user.getUuid());
         List<AddressDto> addressDto =
             allAddress.stream()
                 .filter(a -> a.getAddressStatus() != AddressStatus.DELETED)
                 .map(a -> modelMapper.map(a, AddressDto.class))
                 .collect(Collectors.toList());
         userProfileDto.setAddressDto(addressDto);
+        userProfileDto.setBotList(botList);
         return userProfileDto;
     }
 
+    private List<Bot> getListOfBots(String uuid) {
+        List<Bot> botList = new ArrayList<>();
+        EnumSet<BotType> botTypeEnumSet = EnumSet.allOf(BotType.class);
+        for (BotType botType : botTypeEnumSet) {
+            botList.add(new Bot(botType.name(), createLink(botType.name(), uuid)));
+        }
+        return botList;
+    }
+
+    private String createLink(String type, String uuid) {
+        StringBuilder stringBuilder = new StringBuilder();
+        if (type.equals("TELEGRAM")) {
+            stringBuilder.append(TELEGRAM_PART_1_OF_LINK)
+                .append(telegramBotName).append(TELEGRAM_PART_3_OF_LINK)
```

Comment:
instead of dealing with "parts", use String.format

Reply:
resolved